romancal/tessel_match/tessel_match.py

```diff
@@ -0,0 +1,291 @@
+"""
+This module determines which tessel files overlap with the given image.
+
+Currently this assumes that the sky projected borders of all images are straight.
+"""
+
+import os
+import os.path
+
+import asdf
+import gwcs.wcs as wcs
+import numpy as np
+import spherical_geometry.polygon as sgp
+import spherical_geometry.vector as sgv
+from matplotlib import pyplot as plt
+
+RAD_TO_ARCSEC = 180.0 / np.pi * 3600.0
+
+plt.ion()
```

Comment:
It feels like the matplotlib dependencies should be limited to the plotting functions?  Presently matplotlib is only an optional dependency of romancal, which is included if you also want to be able to build the docs.

Reply:
I am inclined to move that stuff into a separate module. I do think it has some use early on for diagnosing problems and getting a good feel for what is going on. And it should be optional I agree.